rust-install/src/component/package.rs

```diff
@@ -77,6 +99,68 @@ impl Package for DirectoryPackage {
 	}
 }
 
+// On Unix we need to set up the file permissions correctly so
+// binaries are executable and directories readable.
```

Comment:
For the time being it is, yes. There is a [bug with Rust's tarballs](https://github.com/rust-lang/rust/issues/25479) that causes them not to extract with the correct permissions. I don't know how to fix it, but even if we do it will take some time to roll out.